lightning/src/routing/scoring.rs

```diff
@@ -678,25 +795,103 @@ impl<L: Deref<Target = u64>, T: Time, U: Deref<Target = T>> DirectedChannelLiqui
 			} else {
 				let negative_log10_times_2048 =
 					approx::negative_log10_times_2048(numerator, denominator);
-				self.combined_penalty_msat(amount_msat, negative_log10_times_2048, params)
+				Self::combined_penalty_msat(amount_msat, negative_log10_times_2048,
+					params.liquidity_penalty_multiplier_msat,
+					params.liquidity_penalty_amount_multiplier_msat)
+			}
+		};
+
+		if params.historical_liquidity_penalty_multiplier_msat != 0 ||
+		   params.historical_liquidity_penalty_amount_multiplier_msat != 0 {
+			// If historical penalties are enabled, calculate the penalty by walking the set of
+			// historical liquidity bucket (min, max) combinations (where the min is less than the
```

Comment:
Ah, then it may be enough to make explicit that we're talking about the indices?:
```suggestion
			// historical liquidity bucket index (min_idx, max_idx) combinations (where min_idx < max_idx
```